geode-core/src/main/java/org/apache/geode/internal/cache/PartitionedRegion.java

```diff
@@ -9990,7 +9998,56 @@ public EntryExpiryTask getEntryExpiryTask(Object key) {
     return br.getEntryExpiryTask(key);
   }
 
+  PartitionRegionConfig updatePRNodeInformation() {
+    return updatePartitionRegionConfig(prConfig -> {
+      CacheLoader cacheLoader = basicGetLoader();
+      CacheWriter cacheWriter = basicGetWriter();
+      PartitionRegionConfig newConfig = null;
+      if (prConfig != null) {
+        newConfig =
+            new PartitionRegionConfig(prConfig.getPRId(), prConfig.getFullPath(),
```

Comment:
The new `PartitionRegionConfig` may not have all the information that `prConfig` has. i.e. it is possible that the `newConfig` is not equivalent to `prConfig` before updating the loader and writer bits.

Reply:
We looked at this and we changed it to modify the original prConfig instead of creating a new copy. We think this should address the danger of creating an inaccurate update of the prConfig.